protocol.md

```diff
@@ -0,0 +1,117 @@
+# CDBA protocol description
+
+## General stuff:
+
+### cdba operates over ssh
+
+### Client:
+* `~/.ssh/config` is effectively the 'real' config on the client side
+* The connection is established by executing:
+```sh
+# serverbinaryname defaults to 'cdba-server'
+/usr/bin/ssh user@server serverbinaryname
+```
+* ssh `stdout` receives `messages` and the client is expected to handle them instantly after receiving, until `EOF`
+* ssh `stderr` receives `status updates` (blue text) and the client is expected to handle them instantly after receiving, until `EOF`
+
+### Server:
+* The server enumerates all of the connected devices, based on a configuration file
+* Client messages are received over `stdin`
+* Messages are sent to the clients over `stdout`, until the connection is closed (or drops)
+* The server opens a tty to the serial port on the device and forwards the output via `MSG_CONSOLE`
+
+## Supported messages
+`C - client | S - server`
+
+
+### 1 - MSG_SELECT_BOARD [C -> S]
+> ```data = string board name | len = sizeof(data) ```
+>
+> Select the board to operate on.
+>
+> Return [**S**]: Echo the command back with `len = 0`.
+
+### 2 - MSG_CONSOLE [C <-> S]
+> ```data = string_buffer | len = sizeof(data) ```
+> Send TTY I/O.
+>
+> Return [**C**/**S**]: Echo the command back with `len = sizeof(buf), data = buf`.
```

Comment:
The way you describe hte protocol looks almost like a request/response. But at least MSG_CONSOLE isn't a request/response thing, is it?